src/Type/Php/RegexGroupParser.php

```diff
@@ -227,6 +220,29 @@ private function walkRegexAst(
 		}
 	}
 
+	private function allowConstantTypes(
+		string $patternModifiers,
+		bool $repeatedMoreThanOnce,
+		RegexCapturingGroup|RegexNonCapturingGroup|null $parentGroup,
+	): bool
+	{
+		if (str_contains($patternModifiers, 'i')) {
```

Comment:
see https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/v3.61.1/src/DocBlock/TypeExpression.php#L58 regex, modifiers does not have to be global

Reply:
most helpful would be a issue report with a reproducing snippet.

but I am already working on this one, so thanks for the heads up.